torch/csrc/jit/pickle.cpp

```diff
@@ -0,0 +1,71 @@
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <ATen/core/ivalue.h>
+#include <torch/csrc/jit/pickler.h>
+
+
+namespace torch {
+namespace jit {
+
+TORCH_API std::string pickle(
+    const std::vector<IValue>& ivalues,
+    std::vector<at::Tensor>* tensor_table) {
+  std::stringstream ss;
+  Pickler pickler(ss, tensor_table);
+
+  if (tensor_table == nullptr) {
+    // No tensor table provided, so tensors will be stored directly in the blob.
+    // Add torch.save metadata so these tensors can be de-serialized later
+    pickler.pushMetadata();
+  }
+
+  pickler.start();
+
+  bool wrap_in_tuple = ivalues.size() > 0;
```

Comment:
how does unpickler know whether there were multiple IValues or a single IValue which happened to be a tuple? it might be safer to just always wrap into a tuple